docs/rfcs/sync-manager.md

```diff
@@ -0,0 +1,171 @@
+# Sync manager
+
+We've identified the need for a "sync manager" (although are yet to identify a good name for it!) This manager will be responsible for managing "global sync state" and coordinating each engine. For the purposes of this, we are including the "clients" engine in this global state.
```

Comment:
Interesting that the clients engine is being considered, although I guess this is the only sane place for it.